sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubAsyncClient.java

```diff
@@ -0,0 +1,1069 @@
+        if (sendStoppedEvent) {
+            tryEmitNext(stoppedEventSink, new StoppedEvent());
+        }
+
+        groupMessageEventSink.emitComplete(
+            emitFailureHandler("Unable to emit Complete to groupMessageEventSink"));
+        groupMessageEventSink = Sinks.many().multicast().onBackpressureBuffer(Queues.SMALL_BUFFER_SIZE, false);
+
+        serverMessageEventSink.emitComplete(
+            emitFailureHandler("Unable to emit Complete to groupMessageEventSink"));
+        serverMessageEventSink = Sinks.many().multicast().onBackpressureBuffer(Queues.SMALL_BUFFER_SIZE, false);
+
+        connectedEventSink.emitComplete(
+            emitFailureHandler("Unable to emit Complete to connectedEventSink"));
+        connectedEventSink = Sinks.many().multicast().onBackpressureBuffer(Queues.SMALL_BUFFER_SIZE, false);
+
+        disconnectedEventSink.emitComplete(
+            emitFailureHandler("Unable to emit Complete to disconnectedEventSink"));
+        disconnectedEventSink = Sinks.many().multicast().onBackpressureBuffer(Queues.SMALL_BUFFER_SIZE, false);
+
+        stoppedEventSink.emitComplete(
+            emitFailureHandler("Unable to emit Complete to disconnectedEventSink"));
+        stoppedEventSink = Sinks.many().multicast().onBackpressureBuffer(Queues.SMALL_BUFFER_SIZE, false);
+
+        rejoinGroupFailedEventSink.emitComplete(
+            emitFailureHandler("Unable to emit Complete to rejoinGroupFailedEventSink"));
+        rejoinGroupFailedEventSink = Sinks.many().multicast().onBackpressureBuffer(Queues.SMALL_BUFFER_SIZE, false);
+
+        ackMessageSink.emitComplete(emitFailureHandler("Unable to emit Complete to ackMessageSink"));
+        ackMessageSink = Sinks.many().multicast().onBackpressureBuffer(Queues.SMALL_BUFFER_SIZE, false);
+
+        updateLogger(applicationId, null);
+    }
+
+    private void updateLogger(String applicationId, String connectionId) {
+        logger = new ClientLogger(WebPubSubAsyncClient.class,
+            LoggingUtils.createContextWithConnectionId(applicationId, connectionId));
+        loggerReference.set(logger);
+    }
+
+    private static final class StopReconnectException extends RuntimeException {
+        private StopReconnectException(String message) {
+            super(message);
+        }
+    }
+
+    private static final class SequenceAckId {
+
+        private final AtomicLong sequenceId = new AtomicLong(0);
+        private final AtomicBoolean updated = new AtomicBoolean(false);
+
+        private void clear() {
+            sequenceId.set(0);
+            updated.set(false);
+        }
+
+        private Long getUpdated() {
+            if (updated.compareAndSet(true, false)) {
+                return sequenceId.get();
+            } else {
+                return null;
+            }
+        }
+
+        private void update(long id) {
+            long previousId = sequenceId.getAndUpdate(existId -> Math.max(id, existId));
+
+            if (previousId < id) {
```

Comment:
Seems we always set `updated` as if we get an id < current, we should response the largest sequenceAck to avoid service keep sending duplicated messages